MariaDB/server
# MDEV-32462: mysql_upgrade -s still checks for non system tables

## How can this PR be tested?
1. Create configuration file
```bash
$ cat ~/.my105.cnf
[mariadb]
datadir=/tmp/10.5
lc_messages_dir=/home/anel/GitHub/mariadb/server/build/10.5/sql/share
general_log=1
general_log_file=patch.log
plugin_load_add=type_mysql_json.so
plugin_dir=/home/anel/GitHub/mariadb/server/build/10.5/plugin/type_mysql_json/
```

2. Make sure library exists
```bash
$ ls /home/anel/GitHub/mariadb/server/build/10.5/plugin/type_mysql_json/
CMakeFiles  cmake_install.cmake  Makefile  type_mysql_json.so
```

3. Start manually server from source
```bash
$ ./scripts/mariadb-install-db --srcdir=../../10.5 --defaults-file=~/.my105.cnf
$ ./sql/mysqld --defaults-file=~/.my105.cnf
```

4. Copy MySQL 5.7 tables to `test` (user) and `mysql` (system) databases in MariaDB
```bash
#  from source
$ cp mysql-test/std_data/mysql_json/mysql_json_test.* /tmp/10.5/mysql/ && cp mysql-test/std_data/mysql_json/mysql_json_test.* /tmp/10.5/test/
```

5. Validate results
 ```sql
 MariaDB [mysql]> show tables from test;
+-----------------+
| Tables_in_test  |
+-----------------+
| mysql_json_test |
+-----------------+
1 row in set (0.001 sec)

MariaDB [mysql]> show tables from mysql like '%json%';
+--------------------------+
| Tables_in_mysql (%json%) |
+--------------------------+
| mysql_json_test          |
+--------------------------+
1 row in set (0.001 sec)
MariaDB [mysql]> select * from mysql.mysql_json_test;
ERROR 1707 (HY000): Table rebuild required. Please do "ALTER TABLE `mysql.mysql_json_test` FORCE" or dump/reload to fix it!
MariaDB [mysql]> select * from test.mysql_json_test;
ERROR 1707 (HY000): Table rebuild required. Please do "ALTER TABLE `test.mysql_json_test` FORCE" or dump/reload to fix it!
```
6. Execute `mariadb_upgrade --force`
<details>
  <summary>
    <h2>
      mariadb_upgrade --force
    </h2>
  </summary>
<pre>
$ sudo ./client/mariadb-upgrade -s --socket=/tmp/mysql.sock --port=3306 --verbose --verbose --verbose --force
Looking for 'mariadb' as: ./client/mariadb
Looking for 'mariadb-check' as: ./client/mariadb-check
The --upgrade-system-tables option was used, user tables won't be touched.
Phase 1/7: Checking and upgrading mysql database
Running 'mariadb-check' with connection arguments: --port='3305' --socket='/run/mysqld/mysqld.sock' --socket='/tmp/mysql.sock' --port='3306' 
# Connecting to localhost...
# Disconnecting from localhost...
Processing databases
mysql
CHECK TABLE `column_stats`  FOR UPGRADE
mysql.column_stats                                 OK
CHECK TABLE `columns_priv`  FOR UPGRADE
mysql.columns_priv                                 OK
CHECK TABLE `db`  FOR UPGRADE
mysql.db                                           OK
CHECK TABLE `event`  FOR UPGRADE
mysql.event                                        OK
CHECK TABLE `func`  FOR UPGRADE
mysql.func                                         OK
CHECK TABLE `global_priv`  FOR UPGRADE
mysql.global_priv                                  OK
CHECK TABLE `gtid_slave_pos`  FOR UPGRADE
mysql.gtid_slave_pos                               OK
CHECK TABLE `help_category`  FOR UPGRADE
mysql.help_category                                OK
CHECK TABLE `help_keyword`  FOR UPGRADE
mysql.help_keyword                                 OK
CHECK TABLE `help_relation`  FOR UPGRADE
mysql.help_relation                                OK
CHECK TABLE `help_topic`  FOR UPGRADE
mysql.help_topic                                   OK
CHECK TABLE `index_stats`  FOR UPGRADE
mysql.index_stats                                  OK
CHECK TABLE `innodb_index_stats`  FOR UPGRADE
mysql.innodb_index_stats                           OK
CHECK TABLE `innodb_table_stats`  FOR UPGRADE
mysql.innodb_table_stats                           OK
CHECK TABLE `mysql_json_test`  FOR UPGRADE
mysql.mysql_json_test                              Needs upgrade
CHECK TABLE `plugin`  FOR UPGRADE
mysql.plugin                                       OK
CHECK TABLE `proc`  FOR UPGRADE
mysql.proc                                         OK
CHECK TABLE `procs_priv`  FOR UPGRADE
mysql.procs_priv                                   OK
CHECK TABLE `proxies_priv`  FOR UPGRADE
mysql.proxies_priv                                 OK
CHECK TABLE `roles_mapping`  FOR UPGRADE
mysql.roles_mapping                                OK
CHECK TABLE `servers`  FOR UPGRADE
mysql.servers                                      OK
CHECK TABLE `table_stats`  FOR UPGRADE
mysql.table_stats                                  OK
CHECK TABLE `tables_priv`  FOR UPGRADE
mysql.tables_priv                                  OK
CHECK TABLE `time_zone`  FOR UPGRADE
mysql.time_zone                                    OK
CHECK TABLE `time_zone_leap_second`  FOR UPGRADE
mysql.time_zone_leap_second                        OK
CHECK TABLE `time_zone_name`  FOR UPGRADE
mysql.time_zone_name                               OK
CHECK TABLE `time_zone_transition`  FOR UPGRADE
mysql.time_zone_transition                         OK
CHECK TABLE `time_zone_transition_type`  FOR UPGRADE
mysql.time_zone_transition_type                    OK
CHECK TABLE `transaction_registry`  FOR UPGRADE
mysql.transaction_registry                         OK

Repairing tables
REPAIR NO_WRITE_TO_BINLOG TABLE `mysql`.`mysql_json_test` 
mysql.mysql_json_test                              OK
Phase 2/7: Installing used storage engines... Skipped
Phase 3/7: Fixing views... Skipped
Phase 4/7: Running 'mysql_fix_privilege_tables'
Phase 5/7: Fixing table and database names ... Skipped
Phase 6/7: Checking and upgrading tables... Skipped
Phase 7/7: Running 'FLUSH PRIVILEGES'
OK
Running 'mariadb-check' with connection arguments: --port='3305' --socket='/run/mysqld/mysqld.sock' --socket='/tmp/mysql.sock' --port='3306' 
</pre>
</details>

7. Check `general_log_file` (make sure proper query is executed only on `mysql`)
<details>
  <summary>
    <h2>
      General log file
    </h2>
  </summary>
<pre>
 17 Connect	root@localhost on mysql using Socket
		    17 Query	SET SQL_LOG_BIN=0, WSREP_ON=OFF
		    17 Query	SELECT table_comment FROM information_schema.tables WHERE table_schema='mysql' and table_comment LIKE 'Unknown data type: %'
</pre>
</details>

## Basing the PR against the correct MariaDB version

## PR quality check
